File: backend/GameApp.Application/Services/GameServices/GameGetService.cs
using GameApp.Application.UseCases.GameUseCases;
using GameApp.Domain.Entities;
using GameApp.Domain.Repositories;

namespace GameApp.Application.Services.GameServices;

public class GameGetService : GameGetUseCase
{
    private IGameRepository _repo;

    public GameGetService(IGameRepository repo)
    {
        _repo = repo;
    }

    public async Task<Game?> GetGame(Guid id)
    {
        return await _repo.FetchByIdAsync(id);
    }

    public async Task<IEnumerable<Game>> GetAllGames()
    {
        return await _repo.FetchAllAsync();
    }
}

File: frontend/gameapp-frontend/src/application/services/game-services/game-get-service.ts
import type { IGameGetUseCase } from "../../usecases/game-use-cases/game-get-use-case";
import type { IGameRepository } from "../../repositories/igame-repository";
import { Game } from "../../../domain/entities/game";

export class GameGetService implements IGameGetUseCase {
    constructor(private readonly gameRepository: IGameRepository) {}

    async getGame(id: string): Promise<Game> {
        return await this.gameRepository.fetchById(id);
    }

    async getAllGames(): Promise<Game[]> {
        return await this.gameRepository.fetchAll();
    }
}
